mu-plugins/enforce-2fa.php

```diff
@@ -0,0 +1,438 @@
+	if ( ! isset( $_COOKIE[ E2FA_IS_JETPACK_SSO_COOKIE ] ) ) {
+		return false;
+	}
+
+	// Check if the user is logged in with Jetpack SSO.
+	$cookie = sanitize_text_field( wp_unslash( $_COOKIE[ E2FA_IS_JETPACK_SSO_COOKIE ] ) );
+	return wp_validate_auth_cookie( $cookie, 'secure_auth' );
+}
+
+/**
+ * Check if the user is logged in with Jetpack SSO and has 2SA enabled.
+ *
+ * @return bool
+ */
+function e2fa_is_jetpack_sso_two_step() {
+
+	// If the user is not logged in, return early.
+	if ( ! e2fa_is_jetpack_sso() ) {
+		return false;
+	}
+
+	// If the user is not logged in with Jetpack SSO 2SA, return early.
+	if ( ! isset( $_COOKIE[ E2FA_IS_JETPACK_SSO_2SA_COOKIE ] ) ) {
+		return false;
+	}
+
+	// Check if the user is logged in with Jetpack SSO 2SA.
+	$cookie = sanitize_text_field( wp_unslash( $_COOKIE[ E2FA_IS_JETPACK_SSO_2SA_COOKIE ] ) );
+	return wp_validate_auth_cookie( $cookie, 'secure_auth' );
+}
+
+/**
+ * Should Force 2FA.
+ *
+ * @return bool
+ */
+function e2fa_should_force_two_factor() {
+
+	// Fallout, if Localhost.
+	if ( 'local' === wp_get_environment_type() || ( isset( $_SERVER['HTTP_HOST'] ) && in_array( $_SERVER['HTTP_HOST'], array( 'localhost', '127.0.0.1', '[::1]', '192.168.0.1', '192.168.1.1' ), true ) ) ) {
+		return false;
+	}
+
+	// If we're not forcing 2FA, return early.
+	if ( ! class_exists( 'Two_Factor_Core' ) ) {
+		return false;
+	}
+
+	// Check if user have 2FA enabled.
+	if ( apply_filters( 'is_user_using_two_factor', false ) ) {
+		return false;
+	}
+
+	// Don't force 2FA for Jetpack SSO users that have Two-step enabled.
+	if ( e2fa_is_jetpack_sso_two_step() ) {
+		return false;
+	}
+
+	// If it's a request attempting to connect a local user to a WordPress.com user via XML-RPC or REST, allow it through.
+	if ( e2fa_is_jetpack_authorize_request() ) {
+		return false;
+	}
+
+	// Allow custom SSO solutions.
+	if ( e2fa_use_custom_sso() ) {
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * Use Custom SSO.
+ *
+ * @return bool
+ */
+function e2fa_use_custom_sso() {
+
+	$custom_sso_enabled = apply_filters( 'e2fa_use_custom_sso', null );
+	if ( null !== $custom_sso_enabled ) {
+		return $custom_sso_enabled;
+	}
+
+	// Check for OneLogin SSO.
+	if ( function_exists( 'is_saml_enabled' ) && is_saml_enabled() ) {
+		return true;
+	}
+
+	// Check for SimpleSaml.
+	if ( function_exists( '\HumanMade\SimpleSaml\instance' ) && \HumanMade\SimpleSaml\instance() ) {
+		return true;
+	}
+
+	return false;
+}
+
+/**
+ * Check if the current request is a Jetpack REST API request.
+ *
+ * @return bool
+ */
+function e2fa_is_jetpack_authorize_request() {
+
+	return (
+		// XML-RPC Jetpack authorize request
+		// This works with the classic core XML-RPC endpoint, but not
+		// Jetpack's alternate endpoint.
+		defined( 'XMLRPC_REQUEST' ) && XMLRPC_REQUEST
+		&& isset( $_GET['for'] ) && 'jetpack' === $_GET['for']  // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
Please add a reason on why we are disabling this sniff. You can do so like `// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Reason to ignore.`

Reply:
Added 👍